marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/entitySearch/getOpticPlanTest.sjs

```diff
@@ -0,0 +1,104 @@
+const search = require('/MarkLogic/appservices/search/search');
+const test = require("/test/test-helper.xqy");
+const hubTest = require("/test/data-hub-test-helper.sjs");
+const op = require('/MarkLogic/optic');
+
+const queryOptions = '<options xmlns="http://marklogic.com/appservices/search"/>';
+
+function invokeOpticPlanService(schemaName, viewName, limit, structuredQuery, searchText, sortOrder, columns, retries=0) {
+    try {
+        // Need to clear the modules cache or unrelated error could be thrown. Investigating to file a bugtrack.
+        if (retries === 0) {
+            xdmp.moduleCacheClear();
+        }
+        return fn.head(xdmp.invoke(
+            "/data-hub/5/data-services/entitySearch/getOpticPlan.sjs",
+            {
+                schemaName,
+                viewName,
+                limit,
+                structuredQuery,
+                searchText,
+                queryOptions,
+                sortOrder: xdmp.toJSON(sortOrder).root,
+                columns: Sequence.from(columns)
+            }
+        ));
+    } catch (e) {
+        if (e.code === 'SQL-TABLEREINDEXING' || e.name === 'SQL-TABLEREINDEXING' && retries < 20) {
+            if (retries === 0) {
+                xdmp.sleep(1000);
+            } else {
+                xdmp.sleep(500);
+            }
+            xdmp.log(`Optic retry: ${retries}`);
+            return invokeOpticPlanService(schemaName, viewName, limit, structuredQuery, searchText, sortOrder, columns, ++retries);
+        } else {
+            throw e;
+        }
+    }
+}
+
+function testValidAscendingPlan() {
+    const structuredQuery = xdmp.quote(search.parse(''));
+    const plan = invokeOpticPlanService(
+        'EntitySearchEntity',
+        'EntitySearchEntity', 10,
+        structuredQuery, '',
+        [{'name':'searchEntityProp1', 'ascending':true}],
```

Comment:
@ryanjdew  @rahulvudutala  Are the names here based on property names in the entity model, or constraint names in search options?